src/tests/unit/tests/DetailsView/components/save-assessment-button.test.tsx

```diff
@@ -36,71 +44,103 @@ describe('SaveAssessmentButton', () => {
-        it('snapshot of dialog', () => {
-            expect(wrapper.getElement()).toMatchSnapshot();
-        });
+            it('dialog is visible', () => {
+                wrapper = render(<SaveAssessmentButton {...propsStub} />);
 
-        it('dialog is visible', () => {
-            expect(wrapper.find(Dialog).props().hidden).toEqual(false);
-        });
+                expect(getMockComponentClassPropsForCall(Dialog, 2).hidden).toEqual(false);
+            });
 
-        it('dialog is hidden (dismissed) when "got it" button is clicked', () => {
-            wrapper.find(PrimaryButton).simulate('click');
-            expect(wrapper.find(Dialog).props().hidden).toEqual(true);
-        });
+            it('dialog is hidden (dismissed) when onDismiss is called', () => {
+                wrapper = render(<SaveAssessmentButton {...propsStub} />);
 
-        it('dialog is hidden (dismissed) when onDismiss is called', () => {
-            wrapper.find(Dialog).prop('onDismiss')();
-            expect(wrapper.find(Dialog).props().hidden).toEqual(true);
+                act(() => {
+                    getMockComponentClassPropsForCall(Dialog, 2).onDismiss();
+                });
+                expect(getMockComponentClassPropsForCall(Dialog, 3).hidden).toEqual(true);
+            });
         });
+        let wrapper: RenderResult;
+        describe('interaction', () => {
+            beforeEach(() => {
+                useOriginalReactElements('@fluentui/react', [
+                    'Dialog',
+                    'DialogFooter',
+                    'Stack',
+                    'Checkbox',
+                    'PrimaryButton',
+                ]);
 
-        it('when "dont show again" box is clicked, set the showSaveAssessmentDialog user config state to `false`', () => {
-            // The "Don't show again" checkbox logic is inverted
-            const checkbox = wrapper.find('StyledCheckboxBase');
-            // Check "Don't show again" = true
-            checkbox.simulate('change', null, true);
-            // showSaveAssessmentDialog = false ("Enable the dialog" = false)
-            userConfigMessageCreatorMock.verify(
-                x => x.setSaveAssessmentDialogState(false),
-                Times.atLeastOnce(),
-            );
-        });
+                wrapper = render(<SaveAssessmentButton {...propsStub} />);
+                fireEvent.click(wrapper.getByRole('link'));
+            });
 
-        it('should call saveAssessment on click', async () => {
-            assessmentActionMessageCreatorMock.verify(
-                x => x.saveAssessment(eventStub),
-                Times.atLeastOnce(),
-            );
+            it('dialog is hidden (dismissed) when "got it" button is clicked', async () => {
+                wrapper = render(<SaveAssessmentButton {...propsStub} />);
+
+                await userEvent.click(
+                    wrapper.container.querySelector('mock-customizedprimarybutton'),
+                );
```

Comment:
Because we're fully rendering the button, we need to look for semantic elements.
```suggestion
                await userEvent.click(wrapper.getByRole('button'));
```